core/controller/src/main/scala/whisk/core/controller/Authenticate.scala

```diff
@@ -79,4 +74,25 @@ trait Authenticate {
             Future.successful(None)
         }
     }
+
+  def validateCertificate(name: Option[String])(implicit transid: TransactionId): Future[Option[Identity]] = {
+      name flatMap { entityName =>
+          val name = entityName.substring("CN=".length, entityName.length)
+          Try {
+              //It also supports getting Identity by subject
+              val future = Identity.get(authStore, EntityName(name)) map { result =>
```

Comment:
What's the best way?

Answer: i think first should clear currently model mapping (subject:namespace=1:n, namespace:auth key=1:1), so `through auth key` can get unique Identity.
So as you said below we can also send `namespace` by Httpheader from cli to nginx to controller

```
Current auth key auth: auth key ------------>unique Identity object
client certificate auth : namespace(pass by Header) + subject(saved in `CN`)---->unique Identity object
```